packages/components/src/components/lifecycleValidation.spec.ts

```diff
@@ -22,6 +23,17 @@ const tagNamesPublicWithProps = TAG_NAMES.filter(
 const tagNamesPublicWithoutProps = TAG_NAMES.filter(
   (tagName) => !getComponentMeta(tagName).isInternal && !getComponentMeta(tagName).props
 );
+const tagNamesWithPropsOfTypeObject = TAG_NAMES.filter((tagName) => {
+  return (
+    getComponentMeta(tagName).breakpointCustomizableProps ||
+    (getComponentMeta(tagName).allowedPropValues &&
+      Object.values(getComponentMeta(tagName).allowedPropValues)
+        .map((prop) => {
+          return typeof prop === 'object' && !Array.isArray(prop);
```

Comment:
Might be worth commenting what the array check means and that we might have an issue there in the future if a prop would accept array values.